### DISCUSSION TITLE
Unable to access syslog within a job

### DISCUSSION THREAD

**Discussion by yugr** [community/community#25765] [Actions]
<p>Hi,</p>
<p>I’m trying to access /var/log/syslog in my job via</p>
<pre><code class="lang-auto">if test -r /var/log/syslog; then
  cat /var/log/syslog
fi
</code></pre>
<p>but get an error, presumably because access is blocked by CI:</p>
<pre><code class="lang-auto">+ test -r /var/log/syslog
+ cat /var/log/syslog
make: *** [Makefile:54: check] Error 1
Error: Process completed with exit code 2.
</code></pre>
<p>Here’s a prooflink: <a href="https://github.com/yugr/sortcheck/runs/4684261917?check_suite_focus=true#step:3:53" class="inline-onebox" rel="noopener nofollow ugc">Detect whether syslog is readable. · yugr/sortcheck@50532d2 · GitHub</a></p>
<p>Has anyone experienced something like this and maybe found a solution?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I had to dig around in your code a bit, but that looks more like the <code>grep</code> at the end of this pipeline fails:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/yugr/sortcheck/blob/03c9b8a31f9d72ae9cb224e93efe7fcecc4238b4/test/test.sh#L32-L36">
  <header class="source">

      <a href="https://github.com/yugr/sortcheck/blob/03c9b8a31f9d72ae9cb224e93efe7fcecc4238b4/test/test.sh#L32-L36" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/yugr/sortcheck/blob/03c9b8a31f9d72ae9cb224e93efe7fcecc4238b4/test/test.sh#L32-L36" target="_blank" rel="noopener">yugr/sortcheck/blob/03c9b8a31f9d72ae9cb224e93efe7fcecc4238b4/test/test.sh#L32-L36</a></h4>



  <pre class="onebox">    <code class="lang-sh">
      <ol class="start lines" start="32" style="counter-reset: li-counter 31 ;">
          <li>if test -r /var/log/syslog; then</li>
          <li>  cat /var/log/syslog</li>
          <li>else</li>
          <li>  journalctl -q</li>
          <li>fi | grep a.out.*qsort</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Unfortunately the exit code reported by make is the one of your script, not <code>grep</code>, but my first guess is that it simply didn’t find any matching lines.</p>

**Comment by yugr** [comment]
<p>OMG <a class="mention" href="/airtower-luna">@airtower-luna</a> I was so silly to blame GH here. Thanks for spotting this!</p>
